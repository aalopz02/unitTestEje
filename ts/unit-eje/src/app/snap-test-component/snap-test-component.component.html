<p>snap-test-component works!</p>
<div>
  <div *ngIf="item?.photoUrl as photoUrl; else noImage"> <img [src]="photoUrl" alt=""> </div>
  <ng-template #noImage><p>noImage</p></ng-template>
</div>
