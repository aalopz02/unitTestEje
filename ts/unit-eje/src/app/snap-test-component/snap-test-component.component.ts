import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-snap-test-component',
  templateUrl: './snap-test-component.component.html',
  styleUrls: ['./snap-test-component.component.css']
})
export class SnapTestComponentComponent implements OnInit {

  constructor() { }
  item:any;
  ngOnInit(): void {
  }

}
