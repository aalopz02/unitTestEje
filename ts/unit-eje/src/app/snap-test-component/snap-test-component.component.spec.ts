import { ComponentFixture, TestBed } from '@angular/core/testing';
import { SnapTestComponentComponent } from './snap-test-component.component';


describe('SnapTestComponentComponent', () => {
  let component: SnapTestComponentComponent;
  let fixture: ComponentFixture<SnapTestComponentComponent>;
  const itemWithImage = {
    name: 'Corgi',
    photoUrl: 'https://some-image',
  };

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ SnapTestComponentComponent ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(SnapTestComponentComponent);
    component = fixture.componentInstance;
    //component.item = itemWithImage;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render with image', () => {
    expect(fixture).toMatchSnapshot(); 
  });
});
