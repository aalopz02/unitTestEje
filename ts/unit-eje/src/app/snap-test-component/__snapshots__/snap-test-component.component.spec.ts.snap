// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapTestComponentComponent should render with image 1`] = `
<app-snap-test-component
  item={[Function Object]}
>
  <p>
    snap-test-component works!
  </p><div>
    
    <div>
      <img
        alt=""
        src="https://some-image"
      />
    </div>
    
  </div>
</app-snap-test-component>
`;
